visiont/scripts/data_gen.py

```diff
@@ -0,0 +1,75 @@
+from tqdm import tqdm
+import numpy as np
+
+from torch.utils.data import DataLoader
+from torchvision.transforms import (
+    Compose,
+    ToTensor,
+    Resize,
+    RandomCrop,
+    RandomHorizontalFlip,
+    RandomApply,
+    RandomGrayscale,
+    ColorJitter,
+    GaussianBlur,
+    RandomRotation,
+    RandomErasing,
+)
+from skimage.io import imsave
+
+from visiont.scripts.train import ImageDirectory, Convert
+
+
+def save_samples(image1, image2, idx, path):
+    def save_tensor_to_image(fpath, img):
+        img = img.squeeze(0)
+        arr = img.cpu().numpy()
+        arr = (np.transpose(arr, (1, 2, 0)) * 255).astype(np.uint8)
+        imsave(fpath.as_posix(), arr)
+
+    save_tensor_to_image(path.joinpath(f"{idx:06}-1.jpg"), image1)
+    save_tensor_to_image(path.joinpath(f"{idx:06}-2.jpg"), image2)
+
+
+def main(args):
+
+    # Just a list to try out different augmentation ideas here
+    # Bring learnings into visiont.script.train.main
+    transform = Compose(
+        [
+            Convert("RGB"),
+            Resize(1024),
+            RandomCrop(640),
```

Comment:
This has to be in line with how we do it in train, no?

I think a better idea is to have a helper function which writes input1, input2 tensors in the train loop for debugging purpose. If we use this separate generate tool, then it's easy to have them out of sync and dump one thing but we train on another thing.

Reply:
keeping larger size to have a better view of images

Reply:
> This has to be in line with how we do it in train, no?
> 
> I think a better idea is to have a helper function which writes input1, input2 tensors in the train loop for debugging purpose. If we use this separate generate tool, then it's easy to have them out of sync and dump one thing but we train on another thing.

I think we could publish training samples to tensboardX